clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -440,6 +443,10 @@ private synchronized void resetJoinGroupFuture() {
             disableHeartbeatThread();
 
             state = MemberState.REBALANCING;
+            // a rebalance can be triggered consecutively if the previous one failed,
+            // in this case we would not update the start time.
+            if (lastRebalanceStartMs == -1L)
```

Comment:
Wonder if it would be useful to add a "failed rebalances" metric?

Reply:
Good point, will add one.